Core Lib Documentation: `Lib` module

TAdev0 opened the pull request.

enitrat commented on corelib/src/lib.cairo:
````diff
@@ -177,15 +189,23 @@ impl Felt252MulEq of MulEq<felt252> {
     }
 }
 
+/// Performs a division on `felt252` values.
+///
+/// # Examples
+///
+/// ```
+/// use core::felt252_div;
+///
+/// assert!(felt252_div(4, 2) == 2);
+/// ```
+///
````
what happens if you divide 3 by 2 ?

TAdev0 replied:
weird stuff happens :p

enitrat commented on corelib/src/lib.cairo:
```diff
@@ -210,13 +230,15 @@ impl Felt252TryIntoNonZero of TryInto<felt252, NonZero<felt252>> {
 }
 
 impl Felt252Default of Default<felt252> {
+    /// Returns the default value for a `felt252`, i.e., 0.
```
```suggestion
```

enitrat commented on corelib/src/lib.cairo:
```diff
@@ -210,13 +230,15 @@ impl Felt252TryIntoNonZero of TryInto<felt252, NonZero<felt252>> {
 }
 
 impl Felt252Felt252DictValue of Felt252DictValue<felt252> {
+    /// Returns the default value for dictionary `felt252` values, i.e., 0.
```
```suggestion
```

enitrat commented on corelib/src/lib.cairo:
```diff
@@ -227,12 +249,12 @@ impl Felt252Felt252DictValue of Felt252DictValue<felt252> {
 extern fn drop<T>(obj: T) nopanic;
 
-/// Boxes.
+/// Box module.
```
```suggestion
```

remove these doc lines. the module doc is sufficient

enitrat commented on corelib/src/lib.cairo:
```diff
@@ -100,19 +106,23 @@ impl BoolIntoFelt252 of Into<bool, felt252> {
 #[allow(unused_imports)]
 use felt_252::{Felt252One, Felt252Zero};
 
+/// `felt252` is the basic field element used in Cairo.
```
Deserves more doc than that, especially on edge-case behaviors

enitrat commented on corelib/src/lib.cairo:
```diff
@@ -85,8 +91,8 @@ impl BoolPartialEq of PartialEq<bool> {
 
-/// Default values for felt252_dict values.
 impl BoolFelt252DictValue of Felt252DictValue<bool> {
+    /// Returns the default value for dictionary `bool` values, i.e., `false`.
```
```suggestion
```

TAdev0 commented:
@enitrat  i removed all the redundant comments that describe imported modules.

Added a few comments on felt252 type, not sure how deep you wanna go given that everything is in the cairo book

enitrat commented on corelib/src/lib.cairo:
````diff
@@ -201,8 +212,10 @@ impl Felt252Neg of Neg<felt252> {
 ///
 /// assert!(felt252_div(4, 2) == 2);
-/// ```
+/// assert!(felt252_div(4, 3) ==
+/// 1206167596222043737899107594365023368541035738443865566657697352045290673495);
````
document this strange behavior

TAdev0 replied:
done
